File: app/src/main/java/model/DateTime.java
package model;

import androidx.annotation.Keep;

import java.io.Serializable;

import ir.hamsaa.persiandatepicker.api.PersianPickerDate;

@Keep
public class DateTime implements Serializable {

    private PersianPickerDate date;
    private int hour;
    private int minute;

    public PersianPickerDate getDate() {
        return date;
    }

    public void setDate(PersianPickerDate date) {
        this.date = date;
    }

    public int getHour() {
        return hour;
    }

    public void setHour(int hour) {
        this.hour = hour;
    }

    public int getMinute() {
        return minute;
    }

    public void setMinute(int minute) {
        this.minute = minute;
    }

    public String getHourString() {
        return normalizeTime(hour);
    }

    public String getMinuteString() {
        return normalizeTime(minute);
    }

    public String normalizeTime(int clock) {
        return clock < 10 ? "0" + clock : String.valueOf(clock);
    }

    public String getClock() {
        return getHourString() + ":" + getMinuteString();
    }

    public String getPersianDate() {
        return date.getPersianDayOfWeekName() +
                "، " + date.getPersianDay() +
                " " + date.getPersianMonthName() +
                " " + date.getPersianYear();
    }
}

File: app/src/main/java/model/DateTime.kt
package model

import android.os.Parcelable
import androidx.annotation.Keep
import ir.hamsaa.persiandatepicker.api.PersianPickerDate
import kotlinx.parcelize.IgnoredOnParcel
import kotlinx.parcelize.Parcelize

@Keep
@Parcelize
data class DateTime(
    //if field of serializable class not implement Serializable,
    // must be add `transient` keyword to that
    @Transient
    @IgnoredOnParcel
    var date: PersianPickerDate? = null,

    var hour: Int = 0,

    var minute: Int = 0,

    ) : Parcelable {

    val hourString: String
        get() = normalizeTime(hour)

    val minuteString: String
        get() = normalizeTime(minute)

    fun normalizeTime(clock: Int): String {
        return if (clock < 10)
            "0$clock"
        else
            clock.toString()
    }

    val clock: String
        get() = "$hourString:$minuteString"

    val persianDate: String
        get() = date!!.persianDayOfWeekName +
                "، " + date!!.persianDay +
                " " + date!!.persianMonthName +
                " " + date!!.persianYear
}